### ISSUE TITLE
[DRC][ihp-sg13g2] NW.b1: false positive on same-net NWells

### ISSUE THREAD

**Issue by tskaar** [aesc-silicon/gdscheck#6]
### Process / PDK

ihp-sg13g2

### gdscheck version

gdscheck 0.1.2

### Top cell

TOP

### Command line

```shell
gdscheck run -p ihp-sg13g2 -i testcase_nw_same_net.gds -t TOP --suite core --report gdscheck_core.lyp
```

### What's wrong?

False positive (a violation is reported but the layout is correct)

### Rule ID(s)

NW.b1

### Description

Hi!

I noticed that `NW.b1: Min. PWell width between NWell regions (different net) (Note 3) > 1.80 µm` triggers even though the NWell is of the same net. E.g. both of these cases triggers:

<img width="1052" height="791" alt="Image" src="https://github.com/user-attachments/assets/c9137734-6baf-40b6-b409-259a51c5d424" />

While comparing with the PDK KLayout DRC deck, the right hand-side should not trigger `NW.b1`

### GDS file and marker database

[testcase_nw_same_net.zip](https://github.com/user-attachments/files/31157741/testcase_nw_same_net.zip)

### Confirmations

- [x] I have the right to share this layout publicly (this is a public repository under AGPL-3.0-or-later).
- [x] I tried to reduce the GDS to the smallest reproducer I could.

### Environment

OS: Archlinux x86_64
Compiled from the [AUR](https://aur.archlinux.org/packages/gdscheck)

Which uses:

```
export RUSTUP_TOOLCHAIN=stable
export CARGO_TARGET_DIR=target
cargo build --frozen --release --all-features

```

Cargo version: `cargo 1.95.0 (f2d3ce0bd 2026-03-21)`
Rustc version: `rustc 1.95.0 (59807616e 2026-04-14)`

**Comment by dnltz** [comment]
@tskaar thanks for reporting! I will take a look into this.

**Comment by dnltz** [comment]
@tskaar so this rule was geometry only and had no connectivity included. Pull-request #8 fixes that. Do you have a design you can use to verify this PR?

I also added your test pattern to the internal checks to make sure this rules now works properly.

**Comment by tskaar** [comment]
@dnltz

Had a pMOS pair with a spacing of 1.585 µm, that is no longer (false-positively) triggering, thanks for the quick fix!

Love the speed of the tool!

**Comment by dnltz** [comment]
> Love the speed of the tool!

@tskaar that was the intention for writing this tool! Do you have any other feedback that could help me improve this tool?

**Comment by tskaar** [comment]
> > Love the speed of the tool!
> 
> [@tskaar](https://github.com/tskaar) that was the intention for writing this tool! Do you have any other feedback that could help me improve this tool?

@dnltz  Just recently started using the tool, so got nothing at this moment, will keep reporting issues/feedback if I spot anything!
